compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -3658,10 +3682,12 @@ private void defineFieldsOfClassDef(BLangClassDefinition classDefinition, Symbol
 
     private void defineFieldsOfObjectOrRecordTypeDef(BLangTypeDefinition typeDef, SymbolEnv pkgEnv) {
         NodeKind nodeKind = typeDef.typeNode.getKind();
-        if (nodeKind != NodeKind.OBJECT_TYPE && nodeKind != NodeKind.RECORD_TYPE) {
+        if (nodeKind != NodeKind.RECORD_TYPE) {
```

Comment:
I recall discussing this during the review, but just to verify, are we removing this because object fields are defined elsewhere?

Reply:
Also, method name should be updated?

Reply:
Yes, because with introduce of visitors for types, we can use those visitors to define fields of records and objects.